Life-Pill/pharmacy-pos-frontend
# Resolve CORS issue, Connect Login Endpoint, Implement Order Details Sidebar Search, Fix Cashier Dashboard Search Issue

# Pull Request: Resolve CORS issue, Connect Login Endpoint, Implement Order Details Sidebar Search, Fix Cashier Dashboard Search Issue

## Description
This pull request addresses several key updates and fixes:

1. Resolved the CORS issue to ensure proper communication between frontend and backend services.
2. Connected the login endpoint and performed thorough testing to ensure login functionality is working as expected.
3. Implemented the search functionality in the Order Details Sidebar, allowing users to search by name in the barcode field.
4. Fixed an issue with the search functionality in the Cashier Dashboard to ensure it works correctly.

## Changes Made
- Resolved CORS issue by updating backend CORS configurations.
- Connected the login endpoint in the frontend code and added appropriate error handling.
- Implemented barcode search by name functionality in the Order Details Sidebar component.
- Fixed search functionality in the Cashier Dashboard component to address issues with filtering.

## Manual Testing Done
- Tested login functionality with valid and invalid credentials to ensure proper authentication.
- Tested barcode search by name feature in the Order Details Sidebar with various inputs and edge cases.
- Tested search functionality in the Cashier Dashboard with different search queries and verified correct filtering.

## Related issues
This pull request addresses the following issues:
- Closes #33 

## Author
Tharindu Jayawardhana
Frontend Developer at LifePill
